hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_api_order_book_data_source.py

```diff
@@ -0,0 +1,315 @@
+from collections import defaultdict
+
+import aiohttp
+import asyncio
+import logging
+
+from typing import (
+    Dict,
+    List,
+    Optional, Any,
+)
+
+from hummingbot.connector.derivative.bybit_perpetual import bybit_perpetual_constants as CONSTANTS, bybit_perpetual_utils
+from hummingbot.core.data_type.order_book import OrderBook, OrderBookMessage
+
+from hummingbot.connector.derivative.bybit_perpetual.bybit_perpetual_order_book import BybitPerpetualOrderBook
+from hummingbot.connector.derivative.bybit_perpetual.bybit_perpetual_websocket_adaptor import BybitPerpetualWebSocketAdaptor
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.logger import HummingbotLogger
+
+
+class BybitPerpetualAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    _ORDER_BOOK_SNAPSHOT_DELAY = 60 * 60
+
+    _logger: Optional[HummingbotLogger] = None
+    _trading_pair_symbol_map: Dict[str, Dict[str, str]] = {}
+    _last_traded_prices: Dict[str, Dict[str, float]] = defaultdict(dict)
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._logger is None:
+            cls._logger = logging.getLogger(__name__)
+        return cls._logger
+
+    def __init__(self, trading_pairs: List[str] = None,
+                 domain: Optional[str] = None,
+                 session: Optional[aiohttp.ClientSession] = None):
+        super().__init__(trading_pairs)
+        self._domain = domain
+        self._trading_pairs: List[str] = trading_pairs
+        self._session = session or aiohttp.ClientSession()
+        self._messages_queues: Dict[str, asyncio.Queue] = defaultdict(asyncio.Queue)
+
+    async def _create_websocket_connection(self) -> BybitPerpetualWebSocketAdaptor:
+        """
+        Initialize WebSocket client for UserStreamDataSource
+        """
+        try:
+            ws = await self._session.ws_connect(bybit_perpetual_utils.wss_url(self._domain))
+            return BybitPerpetualWebSocketAdaptor(websocket=ws)
+        except asyncio.CancelledError:
+            raise
+        except Exception as ex:
+            self.logger().network(f"Unexpected error occurred during {CONSTANTS.EXCHANGE_NAME} WebSocket Connection "
+                                  f"({ex})")
+            raise
+
+    @classmethod
+    async def init_trading_pair_symbols(cls, domain: Optional[str] = None):
+        """Initialize _trading_pair_symbol_map class variable
+        """
+        cls._trading_pair_symbol_map[domain] = {}
+
+        endpoint_url = bybit_perpetual_utils.rest_api_url_for_endpoint(CONSTANTS.QUERY_SYMBOL_ENDPOINT, domain)
+
+        async with aiohttp.ClientSession() as client:
+            async with client.get(endpoint_url, params={}) as response:
```

Comment:
Is there an advantage to using a context manager for the requests? If not, I propose going the alternative route for two reasons:

1. It simplifies the API mocking since we don't have to play around with `.__aenter__` and `.__aexit__`.
2. It allows us to use the throttler lock more narrowly (as locks should lock the smallest piece of code possible). I.e.:
```Python
async with aiohttp.ClientSession() as client:
    async with self._throttler.execute_task(endpoint_url):
        response = await client.get(endpoint_url, params={})
    """do stuff with the response that don't need to be under the throttler's lock"""
    if response.status == 200:
        ...
```
Bonus point: it removes one level of indentation